translationRecorder/app/src/main/java/org/wycliffeassociates/translationrecorder/permissions/PermissionsDialogActivity.kt

```diff
@@ -0,0 +1,55 @@
+package org.wycliffeassociates.translationrecorder.permissions
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.view.WindowManager
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import org.wycliffeassociates.translationrecorder.R
+import org.wycliffeassociates.translationrecorder.login.Screen
+import android.provider.Settings
+
+private const val PERMISSIONS_PAGE = 50
+
+class PermissionsDialogActivity : AppCompatActivity() {
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        window.addFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        Screen.lockOrientation(this)
+        createPermissionDialog()
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if(requestCode == PERMISSIONS_PAGE) {
+            finish()
+        }
+    }
+
+    private fun createPermissionDialog() {
+        AlertDialog.Builder(this)
```

Comment:
What do you think, if we show them a dialog window with the link that leads to app settings where they can set permissions? Like that:
```Intent intent = new Intent();
intent.setAction(Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
Uri uri = Uri.fromParts("package", getPackageName(), null);
intent.setData(uri);
startActivity(intent);```